doc/source/kubecluster2.rst

```diff
@@ -0,0 +1,106 @@
+KubeCluster2
```

Comment:
Lets try and frame this as the Dask Operator. `KubeCluster2` is just one part of it. It would be good to mention that the operator gets installed on the cluster and then users can create Dask clusters via the Kubernetes API (kubectl) or the Python API (`KubeCluster2`).